File: Daily/2024-01/2024-01-28/7570.줄세우기.py
import sys; input = sys.stdin.readline
n = int(input())
arr = [*map(int, input().split())]
dp = [0 for _ in range(n+1)]
for i in range(n):
    dp[arr[i]] = max(dp[arr[i]-1] + 1, dp[arr[i]])
print(n-max(dp))

File: Daily/2024-01/2024-01-28/7570.줄세우기.js
const fs = require("fs");
var [n, ...arr] = fs.readFileSync("./dev/stdin").toString().split(/\s/).map(Number);
var dp = new Array(n + 1).fill(0);
for (let i = 0; i < n; i++) {
  dp[arr[i]] = Math.max(dp[arr[i] - 1] + 1, dp[arr[i]]);
}
console.log(n - Math.max(...dp));
